### PR TITLE
ResourceSummary: add possibility to pass additional priority info

### PR DESCRIPTION
**Pull request by suomiy**

### CODE REVIEW

**Comment by suomiy** [comment]
needed for #2034

**Comment by spadgett** [comment]
Why not just make this a child of `ResourceSummary`?

It changes the order, but IMO that's a good thing. That way the order is consistent for all resources.

**Comment by spadgett** [comment]
OK, I see this is specifically to move items to the top. I'm still unsure about this though since it makes pages inconsistent.

**Comment by suomiy** [comment]
This was implemented according to http://openshift.github.io/openshift-origin-design/web-console/knikubevirt/vm-details/vm-details design. 

Other resources do not usually have description so the user will not be surprised by a misplaced field (just a new field will appear).

File: frontend/public/components/utils/details-page.tsx [hunk 1]
```diff
@@ -90,6 +91,7 @@ export type ResourceSummaryProps = {
   showAnnotations?: boolean;
   showTolerations?: boolean;
   podSelector?: string;
+  priorityInfo?: React.ReactNode;
```
Comment (suomiy) [inline frontend/public/components/utils/details-page.tsx]: not sure about the naming though

**Comment by spadgett** [comment]
> This was implemented according to http://openshift.github.io/openshift-origin-design/web-console/knikubevirt/vm-details/vm-details design.

The design is inconsistent with other console pages.

> Other resources do not usually have description so the user will not be surprised by a misplaced field (just a new field will appear).

Many other resources do have descriptions. (Projects, templates, service classes, etc.)

**Comment by spadgett** [comment]
In the 3.x console, we didn't have a general summary component like this. Every details page was a little different than every other, with small inconsistencies from page to page. One of the purposes of this component is to prevent this.

I'd like to come up with a common way to show components with description so that it's consistent. I don't think it should go between name and namespace. Those two together are what uniquely identify the resource in the cluster, so I think when both are shown they should be shown next to each other.

It would be better to pass in an optional description to `ResourceSummary` so that it's shown in the same way in all pages. But again, I'm not sure I agree with the proposed placement.

/hold

**Comment by suomiy** [comment]
@matthewcarleton are we ok with having the description in a different place?

**Comment by matthewcarleton** [comment]
@suomiy @spadgett I'm ok with moving it down to keep consistency with other resources.

**Comment by suomiy** [comment]
ok then, closing

**Comment by spadgett** [comment]
Thanks, and sorry if it seems I'm being picky here. I want to be careful about changes to some of the shared components like `Firehose` and `ResourceSummary` that everyone uses.

We have a lot of developers contributing to console now, and it will be a challenge to keep consistency throughout the UI. I feel like our best way to do this is to enforce it through use of shared components like `PageHeading`, `ResourceSummary`, `StatusIcon`, etc. In general, I don't think we should start down the path of putting in special casing these common components for specific pages if possible.

**Comment by matthewcarleton** [comment]
@spadgett Thanks! I think moving forward we'll be better at avoiding these types of conflicts in our designs now that we have more eyes from the OpenShift side on them prior to development. We are definitely on board with consistency and hope to resolve this types of conflicts earlier on so we don't get hung up in dev.

**Pull request closed without merging**